### DISCUSSION TITLE
Instagram first download not starting

### DISCUSSION THREAD

**Discussion by Bloodjack01** [AAndyProgram/SCrawler#62] [Q&A] [answered]
Hi, I managed to find all the correct hash 1 and 2 , the app ID, the claim and the csrftoken... I think!
I fired up the app and get this in the log :
```
2022-07-30 23:12:11: TDownloader.DownloadData
	Object reference not set to an instance of an object.
2022-07-30 23:12:11: Message: Object reference not set to an instance of an object.
	HelpLink: 
	StackTrace:    at SCrawler.API.Instagram.SiteSettings.get_BaseAuthExists()
	   at SCrawler.API.Instagram.SiteSettings.ReadyToDownload(Download What)
	   at SCrawler.DownloadObjects.TDownloader.DownloadData(Job& _Job, CancellationToken Token)
```

What did I miss ?

**Comment by AAndyProgram** [comment, marked as answer]
Cookies?

**Reply by Bloodjack01** [reply]
Thank you for the reply, yes i'm pretty sure this is it but i'm having issues entering them correctly , I know how to find them but i'm missing something on how to enter them correctly. When I press "+From Google chrome" I get a white field, I tried only the session ID from dev-> Application , and the "bigger version" from the request header find via Network but I either get "value can't be 0" or "incorrect name"
And when I press "+Add cookie" I get an error message :
```
Message: Unable to cast object of type 'PersonalUtilities.Tools.WEB.CookieEditorForm' to type 'PersonalUtilities.Forms.Toolbars.IEditToolbar'.
	HelpLink: 
	StackTrace:    at PersonalUtilities.Tools.WEB.CookieEditorForm.CookieEditorForm_Load(Object sender, EventArgs e)
```

Then a window shows up who makes more sense to my small brain but there are no way to save whatever I'm writing in it.

A small tutorial for the cookies would be gladly appreciated

**Reply by AAndyProgram** [reply]
1. I don't know what this error is. I cannot reproduce this error. I need the full log, not just the message.
2. What operating system are you using?
3. Just read the guide and do exactly what it says, **step by step**. How to set up cookies: https://github.com/AAndyProgram/SCrawler/wiki/Settings#how-to-set-up-cookies

**Reply by Bloodjack01** [reply]
Ok so there is a bit of progress, I was able to enter the cookies, I guess. By that I mean, I used the google chrome way, it saved but I can't check or edit them, still getting the error, full log at the end, Win10. 
Now, still for Instagram, it's half working, The app grabs the tagged posts and the stories but not the "basic" pictures or videos .

[ERROR_Log20220803_110919.txt](https://github.com/AAndyProgram/SCrawler/files/9250340/ERROR_Log20220803_110919.txt)

**Reply by AAndyProgram** [reply]
Reproduced the error. Will be fixed in the next version.

**Reply by Bloodjack01** [reply]
thank you very much! I'll wait then.

**Reply by AAndyProgram** [reply]
Fixed. Please update to the [latest release](https://github.com/AAndyProgram/SCrawler/releases/latest).

**If you like SCrawler, please like the program on [this site](https://alternativeto.net/software/scrawler/about/): https://alternativeto.net/software/scrawler/about/.**
